src/value.rs

```diff
@@ -42,46 +42,86 @@ unsafe impl FromValue for Value {
     }
 }
 
-const NONE: Value = Value(sys::NONE);
-const UNIT: Value = Value(sys::UNIT);
+unsafe impl<'a, T> IntoValue for OCaml<'a, T> {
+    fn into_value(self, _rt: &mut Runtime) -> Value {
+        unsafe { Value::new(self.raw()) }
+    }
+}
+
+unsafe impl<'a, T> FromValue for OCaml<'a, T> {
+    fn from_value(value: Value) -> OCaml<'a, T> {
+        let rt = unsafe { &mut Runtime::recover_handle() };
+        let x: OCaml<T> = unsafe { OCaml::new(rt, value.0) };
+        unsafe { core::mem::transmute(x) }
+    }
+}
+
+unsafe impl<'a, T> IntoValue for OCamlRef<'a, T> {
+    fn into_value(self, _rt: &mut Runtime) -> Value {
+        unsafe { Value::new(self.get_raw()) }
+    }
+}
+
+unsafe impl<'a, T> FromValue for OCamlRef<'a, T> {
+    fn from_value(value: Value) -> OCamlRef<'a, T> {
+        let rt = unsafe { Runtime::recover_handle() };
+        let mut frame = rt.open_frame();
+        let gc = frame.initialize(&[]);
+        let mut x = unsafe { crate::interop::internal::OCamlRawRoot::reserve(gc) };
+        unsafe { core::mem::transmute(x.keep(OCaml::<T>::from_value(value))) }
```

Comment:
👀  this doesn't look very safe. I don't think it makes sense to convert a `Value` into an `OCamlRef` because `Value` doesn't have any information about the runtime handle like `OCaml<'a, T>` does. So producing an `OCamlRef` from a `Value` is not really safe. The other thing is that the frame you create here gets dropped as soon as this method returns, so it is not really doing much.

I would just remove this one.